ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/discovery/internal/PeerDiscoveryController.java

```diff
@@ -156,15 +156,17 @@ public PeerDiscoveryController(
     bootstrapNodes
         .stream()
         .filter(node -> peerTable.tryAdd(node).getOutcome() == Outcome.ADDED)
-        .filter(node -> nodeWhitelist.isPermitted(node))
+        .filter(nodeWhitelist::isPermitted)
         .forEach(node -> bond(node, true));
 
     final long timerId =
         timerUtil.setPeriodic(
             Math.min(REFRESH_CHECK_INTERVAL_MILLIS, tableRefreshIntervalMs),
-            () -> refreshTableIfRequired());
+            this::refreshTableIfRequired);
     tableRefreshTimerId = OptionalLong.of(timerId);
 
+    refreshTableIfRequired();
```

Comment:
I don't think this really makes sense - on start we begin the bonding process with the bootstrap peers.  That is basically a table refresh operation: once we bond, we start requesting peers.

Reply:
We don't send a FIND_NEIGHBOURS request when bonding with a node completes so bonding to the bootstrap nodes doesn't actually perform any discovery.

Reply:
You're right it is - I missed that.  Wonder why it never seems to actually find any nodes...